### DISCUSSION TITLE
Problem with automatically connecting to a WiFi network

### DISCUSSION THREAD

**Discussion by DadoRdz** [dl9rdz/rdz_ttgo_sonde#416] [TTGO firmware]
I use RDZ as a spotter on Lilygo TTGO in permanent operation and it happens to me that it loses connection to the WiFi hotspot and later does not want to connect to it automatically, even though it is the only WiFi network in range and it works properly. I had to come to the location and determine that the RDZ switched to the mode of operation of its own access point at 192.168.4.1, so it was not connected to the cloud through the hotspot. All that had to be done was to start searching again for the WiFi network from the Control menu and after that it immediately connected without any problems to that WiFi hotspot located right next to it. So the RDZ software was not stuck, but for some reason it was not trying to connect to the available network. Of course, resetting the device would immediately solve the problem here, because that would force it to search for configured WiFi networks. Would it be possible to investigate why this problem occurs? Alternatively, the option to set the automatic restart of the RDZ at a given time of the day would be very helpful, as this would certainly avoid problems related to long-term operation.

**Comment by dl9rdz** [comment]
Hard to tell what happened. The software switches to AP mode if it is not able to connect to the hotspot.

Today's devel version has some minor enhancements to the WiFi handling. If you select WiFi mode 1 (that existed before) or 4 (new), it will never switch to AP mode automatically (you can do this manually by pressing a button on the TTGO during the initial WiFi scan).

My recommendation is that you use WiFi mode 1 (if you have potentially multiple networks and want the TTGO to connect to the best), or WiFi mode 4 (if you have only a single network - can even be with hidden SSID in this mode).
